bodyswap.lua

```diff
@@ -149,9 +126,11 @@ function swapAdvUnit(newUnit)
     setNewAdvNemFlags(newNem)
     configureAdvParty(newNem)
     df.global.adventure.player_id = newNem.id
-    activeUnits[newUnitIndex] = oldUnit
-    activeUnits[oldUnitIndex] = newUnit
+    df.global.world.units.adv_unit = newUnit
     oldUnit.idle_area:assign(oldUnit.pos)
+
+    -- +1 is necessary because in adv mode, the window pos is off by 1 from dwarf mode for some reason
+    dfhack.gui.revealInDwarfmodeMap(xyz2pos(newUnit.pos.x+1, newUnit.pos.y+1, newUnit.pos.z), true)
 end
 
 if not dfhack_flags.module then
```

Comment:
the help command (in the print statement below) should be `help bodyswap`, not `bodyswap -help`.